core/lib/zksync_core/src/lib.rs

```diff
@@ -180,6 +182,9 @@ pub enum Component {
     /// Produces input for basic witness generator and uploads it as bin encoded file (blob) to GCS.
     /// The blob is later used as input for Basic Witness Generators.
     BasicWitnessInputProducer,
+    /// Produces input for the TEE verifier and uploads it as bin encoded file (blob) to GCS.
+    /// The blob is later used as input for TEE verifiers.
+    TeeVerifierInputProducer,
```

Comment:
Yeah, not sure if we want to fix the naming of prover/verifier before submitting this PR or we leave it for later revisions.

Reply:
I'm not going to rush to review the code because I'm not deeply familiar with it yet, and I don't have a very strong opinion on the naming convention (it's just [naming](https://xkcd.com/910/), after all). However, I have a feeling there may be a wide consensus on what is a TEE verifier and TEE prover. See examples:
- https://blog.ata.network/tee-prover-with-scroll-0157fbbe65aa
- https://docs.ata.network/tee-overview/tee-prover#design-of-tee-prover

  > There are two main components to the architecture of the TEE Prover with Scroll: 
  >
  >  - **SGX Prover**. An off-chain component that checks that the post-state root matches the existing state root after block execution within the secure enclave, and submits the Proof of Execution (PoE) to the SGX Verifier. 
  >
  >  - **SGX Verifier**. An L1 contract that confirms the correctness of state transition proposed by the SGX Prover. It also verifies the attestation report submitted by the Intel SGX enclave to ensure prover integrity.
- https://github.com/automata-network/sgx-prover
- https://scroll.io/blog/scaling-security#how-will-the-tee-prover-integrate-into-scroll-
  > (...) The prototype SGX prover has already been built and can verify all state transitions of the Scroll Sepolia testnet, where the [code](https://github.com/automata-network/sgx-prover) is fully open-sourced.